src/libraries/Microsoft.Extensions.Primitives/ref/Microsoft.Extensions.Primitives.cs

```diff
@@ -58,8 +59,8 @@ public partial interface IChangeToken
         public bool Equals(Microsoft.Extensions.Primitives.StringSegment other) { throw null; }
         public static bool Equals(Microsoft.Extensions.Primitives.StringSegment a, Microsoft.Extensions.Primitives.StringSegment b, System.StringComparison comparisonType) { throw null; }
         public bool Equals(Microsoft.Extensions.Primitives.StringSegment other, System.StringComparison comparisonType) { throw null; }
-        public override bool Equals(object obj) { throw null; }
-        public bool Equals(string text) { throw null; }
+        public override bool Equals(object? obj) { throw null; }
+        public bool Equals(string? text) { throw null; }
```

Comment:
Doesn't this throw for null?

Reply:
It doesn't allow me to make it non-nullable...

![image](https://user-images.githubusercontent.com/6609929/129456627-660687e4-76b0-4d21-ac5b-9050324a6212.png)

Reply:
Interestingly, in implementation `text` is non-nullable and everything compiles fine

Reply:
> It doesn't allow me to make it non-nullable...

You need to change the type of the interface being implemented to `IEquatable<string?>`.

> in implementation text is non-nullable and everything compiles fine

Yes, because the code never tries to dereference a maybe-null value.  The first thing the code does is check whether it's null and throw if it is.

Reply:
> You need to change the type of the interface being implemented to IEquatable<string?>

Still doesn't allow non-nullable text

Reply:
Sorry, I misread what you'd written.  `IEquatable<T>` is defined to expect `Equals` always allows null (it accepts a `T?`), which is why this is complaining about trying to specify a non-nullable value.  I think the implementation should be changed to return false rather than throwing, and then typed as `string?`, but I'll defer here to @dotnet/area-microsoft-extensions area owners.

Reply:
Made it return false.